src/components/tl/ucp/alltoall/alltoall_pairwise.c

```diff
@@ -89,10 +94,24 @@ ucc_status_t ucc_tl_ucp_alltoall_pairwise_start(ucc_coll_task_t *coll_task)
                                     task->args.dst.info.mem_type);
     }
 
+    /* blocking self send/recv */
+    if ((UCC_TL_UCP_TEAM_CTX(team)->cfg.blocking_a2a_self_copy)) {
+        data_size = (size_t)task->args.src.info.count *
+                    ucc_dt_size(task->args.src.info.datatype);
+        UCPCHECK_GOTO(ucc_tl_ucp_recv_nb((void *)(rbuf + grank * data_size),
+                                          data_size, rmem, grank, team, task),
+                                          task, error);
+        UCPCHECK_GOTO(ucc_tl_ucp_send_nb((void *)(sbuf + grank * data_size),
+                                          data_size, smem, grank, team, task),
+                                          task, error);
+    }
```

Comment:
@bureddy looks like these send/recv to self are still non-blocking. Is that the expected behavior we wanted? don't we need a tight while{} loop here?

Reply:
in xccl also it is non-blocking api but we have a testall() in while loop. . the main thing is we finish self-copy before send/recv other ranks. should be ok. but will change it to  busy loop

Reply:
fixed.